python-threatexchange/threatexchange/cli/experimental_fetch.py

```diff
@@ -36,75 +44,118 @@ def init_argparse(cls, ap) -> None:
+            remaining_attempts = 5
+            while more_to_fetch:
+                try:
+                    result = TE.Net.getThreatUpdates(
+                        privacy_group,
+                        next_page,
+                        start_time=self.start_time,
+                        stop_time=self.stop_time,
+                        threat_types=self.threat_types,
+                        limit=self.limit,
+                    )
+                    if "data" in result:
+                        self._process_indicators(result["data"])
+                    more_to_fetch = "paging" in result and "next" in result["paging"]
+                    next_page = result["paging"]["next"] if more_to_fetch else ""
+                except Exception as e:
+                    remaining_attempts -= 1
+                    print(f"The following error occured:\n{e}")
+                    if type(e) is urllib.error.HTTPError:
+                        print(e.read())
+                    if remaining_attempts > 0:
+                        print(f"\nTrying again {remaining_attempts} more times.")
+                        time.sleep(5)
+                        continue
+                    else:
+                        print(
+                            "\n5 consecutive errors occured, saving state and shutting down!"
+                        )
+                        break
+                remaining_attempts = 5
 
-        for signal_name, signal_type in self.signal_types_by_name.items():
-            if signal_name not in self.counts:
-                continue
-            dataset.store_indicator_cache(signal_type)
-            print(f"{signal_name}: {self.counts[signal_name]}")
+            dataset.store_indicator_cache(self.indicator_store)
+            dataset.set_indicator_checkpoint(
+                privacy_group,
+                self.stop_time,
+                request_time,
+                self.threat_types,
+                next_page,
+            )
 
-        dataset.record_indicator_checkpoint(self.stop_time)
+            (print(f"\nFor privacy group {privacy_group}:"))
+            if remaining_attempts == 5:
+                print("Just like that we are done!")
+            print("\nHere is a summary from this run:")
+            for threat_type in self.counts:
+                print(f"{threat_type}: {self.counts[threat_type]}")
+            print(
+                f"Total: {self.total_count}\n{self.deleted_count} of these were deletes."
+            )
+            print(f"\nThis brings your total dataset to:")
+            total = 0
+            for threat_type in self.indicator_store.state:
+                size = len(self.indicator_store.state[threat_type])
+                total += size
+                print(f"{threat_type}: {size}")
+            print(f"Total: {total}")
+            print(
+                "\nYou can run 'threatexchange -c {config} experimental-fetch --continuation' to continue from the last successful point."
```

Comment:
(not blocking): In general, this tool is much too chatty now. A useful property of CLI tools is their ability to feed into one another from their input and output. In order to get this property, you have to be cognizant of what messages you output on STDOUT